File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.c
bool solve(char* s , int i , int j){
        if(i >= j){
            return true;
        }

        if(s[i] == s[j]){
            return solve(s , i + 1 , j - 1);
        }

        return false;
}

char* longestPalindrome(char* s) {
    int len = strlen(s);
    int Max_len = 0;
    int start = 0;

    for(int i = 0; i < len; i++){
        for(int j = i; j < len; j++){
            if(solve(s , i , j) == true){
                if(j - i + 1 > Max_len){
                    Max_len = j - i + 1;
                    start = i;
                }
            }
        }
    }
    char* target = (char*)malloc((Max_len + 1)* sizeof(char));
    strncpy(target , s + start , Max_len);
    target[Max_len] = '\0';

    return target;
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if not s:
            return ""
        def solve(i: int , j : int) -> tuple[int , int]:
            while i >= 0 and j < len(s) and s[i] == s[j]:
                i -= 1
                j += 1
            return i + 1 , j - 1 - i
        start = 0
        max_len = 0
        for i in range(len(s)):
            start1 , len1 = solve(i , i)
            if len1 > max_len:
                max_len = len1
                start = start1

            start2 , len2 = solve(i , i + 1)
            if max_len < len2:
                max_len = len2
                start = start2

        return s[start : start + max_len]